file-downloader/src/main/java/FileDownloader.java

```diff
@@ -0,0 +1,97 @@
+        options.addOption("o", "output", true, "Output file name");
+        options.addOption("u", "url", true, "URL of the file to download");
+
+        CommandLineParser parser = new DefaultParser();
+
+        try {
+            CommandLine cmd = parser.parse(options, args);
+
+            if (cmd.hasOption("help") || cmd.getOptions().length == 0) {
+                printHelp(options);
+            } else if (cmd.hasOption("url") && cmd.hasOption("output")) {
+                String fileURL = cmd.getOptionValue("url");
+                String outputFile = cmd.getOptionValue("output");
+                int maxRetries = 3; // Maximum number of connection retries
+                downloadFileWithRetries(fileURL, outputFile, maxRetries);
+                logger.info("File downloaded successfully.");
+            } else {
+                logger.error("Missing required options.");
+                printHelp(options);
+            }
+        } catch (ParseException e) {
+            logger.error(e.getMessage());
+            printHelp(options);
+        } catch (IOException | InterruptedException e) {
+            e.printStackTrace();
+            logger.error("File download failed.");
+        }
+    }
+    private static void printHelp(Options options) {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("FileDownloaderWithReconnect", options);
+    }
+
+    public static void downloadFileWithRetries(String fileURL, String outputFile, int maxRetries) throws IOException, InterruptedException {
+        int retries = 0;
+        boolean downloadComplete = false;
+
+        while (retries < maxRetries) {
+            try {
+                downloadFileWithProgressBar(fileURL, outputFile);
+                downloadComplete = true;
+                break; // Download successful, exit loop
+            } catch (IOException e) {
+                logger.info("Connection lost. Retrying...");
+                TimeUnit.SECONDS.sleep(10);//waiting for reconnect (10 sec for retry) until maximum restries (3) will be reached
+                retries++;
+            }
+        }
+
+        if (!downloadComplete) {
+            throw new IOException("Exceeded maximum number of retries");
+        }
+    }
+
+    public static void downloadFileWithProgressBar(String fileURL, String outputFile) throws IOException {
+        URL url = new URL(fileURL);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.connect();
+        int contentLength = connection.getContentLength();
+
+        try (ReadableByteChannel rbc = Channels.newChannel(connection.getInputStream());
+             FileOutputStream fos = new FileOutputStream(outputFile)) {
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            int totalBytesRead = 0;
+
+            while ((bytesRead = rbc.read(ByteBuffer.wrap(buffer))) != -1) {
```

Comment:
Does this logic also handles continue downloading instead of starting from scratch?